front/pages/w/[wId]/assistant/labs/transcripts/index.tsx

```diff
@@ -489,6 +544,65 @@ export default function LabsTranscriptsIndex({
     }
   };
 
+  const handleConnectModjoTranscriptsSource = async () => {
+    try {
+      if (transcriptsConfigurationState.provider !== "modjo") {
+        return;
+      }
+
+      const defaultConfiguration = await getDefaultConfiguration("modjo");
+
+      if (defaultConfiguration) {
+        if (
+          defaultConfiguration.provider !== "modjo" ||
+          !defaultConfiguration.apiKey
+        ) {
+          sendNotification({
+            type: "error",
+            title: "Failed to connect Modjo",
+            description:
+              "Your workspace is already connected to another provider by default.",
+          });
+          return;
+        }
+
+        await saveApiConnection(
+          defaultConfiguration.apiKey,
+          defaultConfiguration.provider,
+          true
+        );
+      } else {
+        if (!transcriptsConfigurationState.apiKey) {
+          sendNotification({
+            type: "error",
+            title: "Modjo API key is required",
+            description: "Please enter your Modjo API key.",
+          });
+          return;
+        }
+        await saveApiConnection(
+          transcriptsConfigurationState.apiKey,
+          transcriptsConfigurationState.provider
+        );
+      }
+
+      sendNotification({
+        type: "success",
+        title: "Modjo connected",
+        description:
+          "Your transcripts provider has been connected successfully.",
+      });
+
+      await mutateTranscriptsConfiguration();
+    } catch (error) {
+      sendNotification({
+        type: "error",
+        title: "Failed to connect Modjo",
+        description: "Could not connect to Modjo. Please try again.",
+      });
+    }
+  };
+
   const handleDisconnectProvider = async () => {
```

Comment:
Any chance we want to start having one component per provider? Given how clutter the code becomes?

Reply:
Will do in a new PR